src/main/java/com/fasterxml/jackson/core/JsonGenerator.java

```diff
@@ -2482,22 +2482,18 @@ public void copyCurrentEvent(JsonParser p) throws IOException
             NumberType n = p.getNumberType();
             if (n == NumberType.INT) {
                 writeNumber(p.getIntValue());
-            } else if (n == NumberType.BIG_INTEGER) {
```

Comment:
I don't think this is necessary: `NumberType.LONG` is only indicated if value fits -- and we do not want to "upgrade" long to `BigInteger` unless it is absolutely necessary.

The reason this matters is that buffering is used quite a lot, for Constructor/Factory method creators etc.

Reply:
I can remove the int/long change - just did this to keep logic consistent with the decimal change

Reply:
For better or worse, integer numbers are much easier: their magnitude can be reliably handled. As such we do want to use the most efficient form since that's what we decode.

But for FP there is that `Double` vs `BigDecimal` distinction which is retained for some backends (binary), but not for others (textual).

So int/FP shouldn't be unified in that sense: there are different constraints, optimal ways.